wolfcrypt/src/port/atmel/atmel.c

```diff
@@ -915,62 +929,72 @@ int atcatls_verify_signature_cb(WOLFSSL* ssl, const byte* sig, unsigned int sigS
 
 static int atcatls_set_certificates(WOLFSSL_CTX *ctx) 
 {
-    #ifndef ATCATLS_TNGTLS_SIGNER_CERT_SIZE
-        #define ATCATLS_TNGTLS_SIGNER_CERT_SIZE 0x208
-    #endif
-    #ifndef ATCATLS_TNGTLS_DEVICE_CERT_SIZE
-        #define ATCATLS_TNGTLS_DEVICE_CERT_SIZE 0x222
-    #endif
-    #ifndef ATCATLS_TNGTLS_CERT_BUFF_SIZE
-        #define ATCATLS_TNGTLS_CERT_BUFF_SIZE (ATCATLS_TNGTLS_SIGNER_CERT_SIZE +\
-                                               ATCATLS_TNGTLS_DEVICE_CERT_SIZE)
-    #endif
-
     int ret = 0;
     ATCA_STATUS status;
-    size_t signerCertSize = ATCATLS_TNGTLS_SIGNER_CERT_SIZE;
-    size_t deviceCertSize = ATCATLS_TNGTLS_DEVICE_CERT_SIZE;
-    uint8_t certBuffer[ATCATLS_TNGTLS_CERT_BUFF_SIZE];
+    size_t signerCertSize=0;
+    size_t deviceCertSize=0;
+    uint8_t *certBuffer;
 
-    /*Read signer cert*/
-    status = tng_atcacert_read_signer_cert(&certBuffer[ATCATLS_TNGTLS_DEVICE_CERT_SIZE],
-                                           &signerCertSize);
+    /* fetch signer cert size */
+    status=tng_atcacert_read_signer_cert(NULL, &signerCertSize);
     if (ATCA_SUCCESS != status) {
-        ret = atmel_ecc_translate_err(ret);
-        return ret;
-    }
-    if (signerCertSize != ATCATLS_TNGTLS_SIGNER_CERT_SIZE) {
         #ifdef WOLFSSL_ATECC_DEBUG
-        printf("signer cert size != ATCATLS_TNGTLS_SIGNER_CERT_SIZE.(%d)\r\n",
-               signerCertSize);
+        printf("Failed reading Signer cert size(0x%x)\r\n", status);
         #endif
         return WOLFSSL_FAILURE;
     }
 
-    /*Read device cert signed by the signer above*/
-    status = tng_atcacert_read_device_cert(certBuffer, &deviceCertSize,\
-     	                  &certBuffer[ATCATLS_TNGTLS_DEVICE_CERT_SIZE]);
+    /* fetch device cert size */
+    status=tng_atcacert_read_device_cert(NULL, &deviceCertSize, NULL);
     if (ATCA_SUCCESS != status) {
+        #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Failed reading device cert size(0x%x)\r\n", status);
+        #endif
+        return WOLFSSL_FAILURE;
+    }    
+    certBuffer=(uint8_t*)XMALLOC(signerCertSize+deviceCertSize, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    if(NULL == certBuffer){
+        #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Failed allocating space for certBuffer\r\n");
+        #endif
+        return WOLFSSL_FAILURE;
+    }
+    
+    /* Read signer cert */
+    status = tng_atcacert_read_signer_cert(&certBuffer[deviceCertSize],
+                                           &signerCertSize);
+    if (ATCA_SUCCESS != status) {
+        #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Error reading signer cert(0x%x)\r\n", status);
+        #endif
+        XFREE(certBuffer,NULL,DYNAMIC_TYPE_TMP_BUFFER);
         ret = atmel_ecc_translate_err(ret);
         return ret;
     }
-    if (deviceCertSize != ATCATLS_TNGTLS_DEVICE_CERT_SIZE) {
+
+    /* Read device cert signed by the signer above */
+    status = tng_atcacert_read_device_cert(certBuffer, &deviceCertSize,\
```

Comment:
Can you remove the trailing `\` on this. Not required for plain C function calls.